go-controller/pkg/ovn/endpoints.go

```diff
@@ -106,7 +60,11 @@ func (ovn *Controller) addEndpoints(ep *kapi.Endpoints) error {
 	for targetPort, ips := range udpPortMap {
 		for _, svcPort := range svc.Spec.Ports {
 			if svcPort.Protocol == kapi.ProtocolUDP && svcPort.TargetPort.IntVal == targetPort {
-				err := ovn.createLoadBalancerVIP(ovn.getLoadBalancer(svcPort.Protocol), svc.Spec.ClusterIP, svcPort.Port, ips, targetPort)
+				if svc.Spec.Type == "NodePort" {
```

Comment:
The python code has a bug in that when svc.Spec.Type is NodePort, it does not create a load-balancer entry for clusterIP. To summarize, when svc.Spec.Type is NodePort, we should create entries in gateway for nodeport as well as for east-west load-balancing.